### ISSUE TITLE
Implementation of all Clustering Algorithms

### ISSUE THREAD

**Issue by DARK-art108** [Dacience/DataScientist#51]
**Is your feature request related to a problem? Please describe.**
Clustering Algorithm

**Describe the solution you'd like**
Implementation of all clustering algorithms in machine learning should be done. A file structure similar to the Classification and Regression should be followed.

**Describe alternatives you've considered**

**Additional context**

**Comment by sidhu1012** [comment]
How clustering would be beneficial? Any practical application?

**Comment by sidhu1012** [comment]
It would be better to use clustering in backend for data cleaning rather than public API.

**Comment by DARK-art108** [comment]
Yeah @sidhu1012 as a backend api implementation of Clustering would be awesome and please check the PR #50 and one also query can we create a implementation of Neural Nets and CNN's

**Comment by sidhu1012** [comment]
> one also query can we create a implementation of Neural Nets and CNN's

I'm trying to figure how to implement neural network. Do you have any idea?

**Comment by DARK-art108** [comment]
As what have got to know till now is that we are implementing each algorithm in a module manner for ex if we can write a raw algorithm of simple ANN than we can import it's module or func anywhere.

**Comment by sidhu1012** [comment]
> As what have got to know till now is that we are implementing each algorithm in a module manner for ex if we can write a raw algorithm of simple ANN than we can import it's module or func anywhere.

But NN takes a lot of tym to train and someone may like to change layers, activation function etc.

So I'm not sure adding NN is a good idea until we have a well defined plan to proceed.

**Comment by DARK-art108** [comment]
Yeah that is also our concern as Neural Nets are not so much time efficient!!So,yeah we should only proceed further when we have a solid structure.

And @sidhu1012 please take a look over the voting classifier PR the Travis is giving err because I have added a new folder similar to regression folder structure once the PR is merged it will no show err on further PR's.

**Comment by sidhu1012** [comment]
> And @sidhu1012 please take a look over the voting classifier PR the Travis is giving err because I have added a new folder similar to regression folder structure once the PR is merged it will no show err on further PR's.


I don't think that's the cause of error, I'll look into it tomorrow as I am a bit busy today.

**Comment by DARK-art108** [comment]
Ok @sidhu1012 yeah sure!!

**Comment by Ak-panchal** [comment]
can u assign this issue

**Comment by sidhu1012** [comment]
> can u assign this issue 

It's more of a discussion. Nothing to work upon yet.
